File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxArea=0;
        while(left<right){
            int area=Math.min(height[left],height[right])*(right-left);
            if(height[right]>height[left]){
                left++;
            }else{
                right--;
            }

            maxArea=Math.max(area,maxArea);
        }

        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int maxArea=INT_MIN;
        while(l<r){
            int area=(r-l)*min(height[r],height[l]);
            maxArea=max(maxArea,area);
            if(height[l]<height[r]){
                l++;
            }else{
                r--;
            }
        }
        return maxArea;
    }
};